### PR TITLE
addons: implement platform filter and only update to tags

### PR DESCRIPTION
**Pull request by Rechi**

### CODE REVIEW

**Comment by Erick555** [comment]
@Rechi [you need this merged first.](https://github.com/flathub/tv.kodi.Kodi/pull/55)

**Comment by wsnipex** [comment]
bot build

**Comment by Rechi** [comment]
@wsnipex ping

File: binary-addons.py [hunk 1]
```diff
@@ -13,29 +14,59 @@
...
         # print some progress
         print(os.path.basename(addon_dir))
-        with open(os.path.join(target_path, "%s.json" % os.path.basename(addon_dir)), 'w') as t:
-            json.dump(addon_dict, t, indent=4)
+
+        t = threading.Thread(target=write_manifest, args=(addon_dir, target_path))
+        t.start()
+        threads.append(t)
+
+    [t.join() for t in threads]
```
Comment (Marko10-000) [inline binary-addons.py]: Is multi-threading* really an improvement/necessary? But it's fine for me.

* Just your subprocesses would run parallel in CPython

File: .gitignore [hunk 2]
```diff
@@ -3,3 +3,80 @@ build-dir/
...
+screensavers.rsxs/
+vfs.libarchive/
+vfs.rar/
+vfs.sftp/
+visualization.fishbmc/
+visualization.goom/
+visualization.projectm/
+visualization.shadertoy/
+visualization.spectrum/
+visualization.waveform/
```
[the remark this answers is not in the document]
Comment (wsnipex) [reply .gitignore]: makes sense for local builds

**Pull request merged**